gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/category/use_case/UpdateCategoryApiOrderUseCase.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.category.use_case;
+
+import io.gravitee.apim.core.UseCase;
+import io.gravitee.apim.core.api.crud_service.ApiCrudService;
+import io.gravitee.apim.core.api.domain_service.ApiAuthorizationDomainService;
+import io.gravitee.apim.core.api.exception.ApiNotFoundException;
+import io.gravitee.apim.core.api.model.Api;
+import io.gravitee.apim.core.api.model.ApiQueryCriteria;
+import io.gravitee.apim.core.category.crud_service.CategoryApiCrudService;
+import io.gravitee.apim.core.category.domain_service.UpdateCategoryApiDomainService;
+import io.gravitee.apim.core.category.domain_service.ValidateCategoryDomainService;
+import io.gravitee.apim.core.category.model.ApiCategoryOrder;
+import io.gravitee.rest.api.service.common.ExecutionContext;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+
+@UseCase
+@RequiredArgsConstructor
+public class UpdateCategoryApiOrderUseCase {
+
+    private final ValidateCategoryDomainService validateCategoryDomainService;
+    private final ApiAuthorizationDomainService apiAuthorizationDomainService;
+    private final UpdateCategoryApiDomainService updateCategoryApiDomainService;
+    private final ApiCrudService apiCrudService;
+    private final CategoryApiCrudService categoryApiCrudService;
+
+    public Output execute(Input input) {
+        // Validate + Get categoryId
+        var categoryId =
+            this.validateCategoryDomainService.validateCategoryIdOrKey(
+                    input.categoryIdOrKey(),
+                    input.executionContext().getEnvironmentId()
+                );
+
+        // Validate that user has access to api
+        this.validateUserHasAccessToApi(input.executionContext(), input.userId(), input.isAdmin(), categoryId, input.apiId());
+
+        // Validate api is associated to category
+        var apiCategoryOrder = this.categoryApiCrudService.get(input.apiId(), categoryId);
+
+        // Change order of api category order
+        this.updateCategoryApiDomainService.changeOrder(apiCategoryOrder, input.newOrder());
```

Comment:
For a UseCase, I generally prefer to have private method with a meaningful name instead of having comments with a what to could look like a unrelated call.